### PR TITLE
chore(ci): add test coverage report

### LINKED ISSUE #4897: CI show run details of flaky tests directly in Jenkins
**Description**

Currently we detect flaky tests, but it is a little cumbersome to get any information about the run of the flaky tests. It would be ideal, if the run information would be displayed directly in Jenkins - as with the other tests.

There is a prototype available at: https://github.com/pihme/jenkinsTestBed/tree/rerun-failed-tests

* This uses a Jenkins Script to harvest the test reports and generates additional test reports for the flaky tests. 
* The names are modified, to make clear that this is the run of a flaky test
* These additional test reports can then be displayed by Jenkins like normal test reports

image

image

This prototype should be applied to Zeebe.

### PR DESCRIPTION
**Pull request by pihme**
## Description

Adds test coverage for Java code.

## Related issues

No related issue

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to backport the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [X] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the release announcement

### CODE REVIEW

**Review by npepinpe — COMMENTED** [review]
Changes look good, just a few questions:

1. Out of curiosity, did the team discuss that we wanted to add code coverage metrics to our build before? I don't remember but if so could you summarize the discussion for me?
1. What's the goal here, longer term? I have my own idea of how to use and interpret code coverage metrics, but I'd like to hear yours :slightly_smiling_face: 

And not a question, but when merged if you could present it to the team as a new tool we have and maybe we can discuss how we want to use it in the team

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 05cc2ac354c676989b42c805cec0ac0fc2229a74
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 05cc2ac354c676989b42c805cec0ac0fc2229a74
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by pihme** [comment]
Won't merge just yet.

Closing this pull request for now, as more work on the branch is needed:
* I am looking for a way to exclude generated sources from the test coverage reports. The generated source line count is substantial and skews the statistics
* On the test coverage branch the problem of flaky tests is much worse, which needs to be investigated why
